Source/Core/InputCommon/ControllerInterface/GameController/GameController.mm

```diff
@@ -0,0 +1,80 @@
+// Copyright 2013 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include <random>
+
+#import <objc/message.h>
+
+#include "InputCommon/ControllerInterface/GameController/GameController.h"
+#include "InputCommon/ControllerInterface/GameController/GameControllerDevice.h"
+
+#include "Common/Logging/Log.h"
+
+namespace ciface::GameController
+{
+static char *s_observer="";
```

Comment:
Why assigning a "" to a ptr?

Reply:
This is to ensure s_observer is an actual pointer to something. Otherwise the pointer will always be 0.

Reply:
Not sure I understand.
I've never allocated a ptr like that.
Does that create a new object and assign it to the pointer?

Reply:
If I understand what this is, you should probably assign "dolphin" to that char array.

Reply:
Why isn't this a constexpr or a const which you then pass as a reference to the callbacks bindings functions?

Reply:
It ensures that there is an empty string in the constant data segment of the binary, and makes the pointer point to it. If this is to be done, the variable should be marked `const char*` instead of just `char*`, because writing to such a string is not allowed.

Reply:
I double checked the manual (https://developer.apple.com/documentation/corefoundation/1543316-cfnotificationcenteraddobserver?language=objc) and it looks like I can completely remove the s_observer variable.
I will remove it and pass nullptr to the callback functions!